File: compiler/bench/simd1.c
// expected: 49995000
// Sum array - pattern GCC can auto-vectorize
#include <stdio.h>
#define SIZE 10000
long arr[SIZE];

int main() {
    for (int i = 0; i < SIZE; i++) arr[i] = i;
    long sum = 0;
    for (int i = 0; i < SIZE; i++) sum += arr[i];
    printf("%ld\n", sum);
    return 0;
}

File: compiler/bench/simd1.fs
\ expected: 49995000
\ Sum array - pattern GCC can auto-vectorize

create arr 80000 allot

: init ( -- ) 10000 0 do i i 8 * arr + ! loop ;
: sum-arr ( -- n ) 0 10000 0 do i 8 * arr + @ + loop ;
: main init sum-arr . cr ;
